src/Azure/Orleans.Transactions.AzureStorage/Storage/AzureTransactionLogStorage.cs

```diff
@@ -359,34 +360,40 @@ private List<CommitRecord> DeserializeCommitRecords(byte[] serializerCommitRecor
 
         public class ArchivalRow : TableEntity
         {
+            public static string MinRowKey = MakeRowKey(long.MinValue);
+            public static string MaxRowKey = MakeRowKey(long.MaxValue);
+
+            public static string MakeRowKey(long firstTransactionId)
+            {
+                return $"arch_{CommitRow.ToRowKey(firstTransactionId)}";
+            }
+
             public ArchivalRow()
             {
             }
 
-            public ArchivalRow(byte[] transactions, long firstTransactionId, long firstLSN, string clusterId, DateTime archivalTime)
+            public ArchivalRow(ClusterOptions clusterOptions, byte[] transactions, long firstTransactionId, long firstLSN)
             {
                 this.Transactions = transactions;
-                this.ArchivalTime = archivalTime;
-                this.ClusterId = clusterId;
+                this.ClusterId = clusterOptions.ClusterId;
                 this.FirstLSN = firstLSN;
                 this.FirstTransactionId = firstTransactionId;
-                PartitionKey = $"{clusterId}${this.ArchivalTime.Ticks}";
-                RowKey = $"{this.ArchivalTime.Ticks}${firstTransactionId}";
+                PartitionKey = clusterOptions.ServiceId;
```

Comment:
As in other cases, prefixing service Id, and either consistently using the CommitRecordPartitionKey, or building the pk.  Current pattern is inconsistent and subject to introducing bugs from future maintenance.